File: Ultimate Goal-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/Pose.java
package org.firstinspires.ftc.teamcode.util;

import android.annotation.SuppressLint;

import org.jetbrains.annotations.NotNull;

public class Pose extends Point implements Cloneable {
    public double heading;

    public Pose(double x, double y, double heading) {
        super(x, y);
        this.heading = heading;
    }

    public Pose(Point p, double heading) {
        this(p.x, p.y, heading);
    }



    @NotNull
    @SuppressLint("DefaultLocale")
    @Override
    public String toString() {
        return String.format("{x: %.3f, y: %.3f, θ: %.3f}", x, y, Math.toDegrees(heading));
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/math/Pose.kt
package org.firstinspires.ftc.teamcode.util.math

import org.firstinspires.ftc.teamcode.control.path.waypoints.Waypoint

data class Pose(
        var p: Point,
        var h: Angle
) {
    val x get() = p.x
    val y get() = p.y
    val cos get() = h.cos
    val sin get() = h.sin
    val hypot get() = p.hypot
    val copy get() = Pose(p, h)

    fun distance(p2: Pose) = p.distance(p2.p)
    fun distance(p2: Waypoint) = p.distance(p2.p)

    val toRawString = String.format("%.2f, %.2f, %.2f", x, y, h.raw)
    override fun toString() = String.format("%.2f, %.2f, %.2f", x, y, h.wrap().deg)
}
